### PR TITLE
Rent mobile lead form changes

### PR DESCRIPTION
**Pull request by tcwilson**
This also fixes a currently failing test on master

[Card](https://rentpath.leankit.com/card/577968668)

### CODE REVIEW

File: packages/react-ui-core/src/Form/__tests__/mocks/theme.js [hunk 1]
```diff
@@ -23,5 +23,6 @@ export default keyMirror([
   'RadioButton-checked',
   'RadioButton-hide',
   'RadioButton-vertical',
+  'RadioButton-hidden',
```
Comment (tadjohnston) [inline packages/react-ui-core/src/Form/__tests__/mocks/theme.js]: There's already a `RadioButton-hide`. Can we remove that one?

File: packages/react-ui-rent/src/LeadForm/Fields/Message.js [hunk 2]
```diff
@@ -25,39 +25,23 @@ export default class Message extends PureComponent {
...
       label,
       ...props
     } = this.props
 
     return (
-      <Field
-        label={this.label}
-        {...props}
-      />
+      <div className={theme.Message}>
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadForm/Fields/Message.js]: Why did you need to remove label here?

File: packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js [hunk 3]
```diff
@@ -1,27 +1,57 @@
...
       PropTypes.object,
       PropTypes.func,
     ]),
+    defaultChecked: PropTypes.bool,
   }
 
   static defaultProps = {
+    theme: {},
     name: 'opt_in_newsletter',
+    id: 'lead_form_opt_in_newsletter',
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js]: If I have multiple lead forms on the page with an `OptInNewsLetter`, how does distinguish between the two? I guess I'm not understanding what the purpose of this `id` is here.

File: packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js [hunk 4]
```diff
@@ -1,27 +1,57 @@
...
+
     return (
-      <Field {...this.props} />
+      <div className={theme.OptInNewsLetter}>
+        <Field
+          id={id}
+          {...props}
+        />
+        {/* eslint-disable jsx-a11y/label-has-for */}
+        <label
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js]: I'm not getting why you moved Label out.  Also, there is a `Label` component you should be using instead.

File: packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js (hunk 3 above)
Comment (tadjohnston) [reply packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js]: If you're trying to use this for `htmlFor`, then please take a look at https://github.com/rentpath/react-ui/blob/master/packages/react-ui-utils/src/randomId.js.  You can look in react-ui-core because it has been used there, especially in regards to `Field`.

File: packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js [hunk 5]
```diff
@@ -1,27 +1,57 @@
...
+    theme: PropTypes.object,
     name: PropTypes.string,
+    id: PropTypes.string,
     type: PropTypes.string,
     label: PropTypes.oneOfType([
       PropTypes.string,
       PropTypes.object,
       PropTypes.func,
     ]),
+    defaultChecked: PropTypes.bool,
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadForm/Fields/OptInNewsLetter.js]: Did you want this to be a controlled or uncontrolled component?

**Comment by lfender6445** [comment]
Are the requirements for Message field differently? I would expect `Message` label to disappear as well

screen shot 2017-12-07 at 3 10 12 pm

**Comment by fredleveau** [comment]
yeah, it would technically disappear as well but users probably don't change the canned message (data?) and therefore would not disappear since there is no new input in the field.

how about:
WHEN user does not edit the canned message
THEN nothing changes and message label remains

WHEN user edits canned message
THEN message label dissapears
AND field behaves in similar manner to other fields

Thoughts?

**Comment by tcwilson** [comment]
@tadjohnston want to give it another look?

File: packages/react-ui-rent/src/LeadModal/LeadModal.js [hunk 6]
```diff
@@ -27,7 +42,12 @@ export default class LeadModal extends Component {
       <Modal
         isOpen={isOpen}
         CloseButton={CloseButton}
+        onClose={onClose}
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadModal/LeadModal.js]: 👍

File: packages/react-ui-rent/src/LeadModal/CloseButton.js [hunk 7]
```diff
@@ -34,6 +36,9 @@ export default class CloseButton extends PureComponent {
           theme.CloseButton
         )}
         {...props}
+        onClick={onClick}
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadModal/CloseButton.js]: You don't need to pass this. The whole point of react-ui-core Modal is so you don't have to do this. See: https://github.com/rentpath/react-ui/blob/master/packages/react-ui-core/src/Modal/ModalBody.js#L41. Passing props through here accomplishes what you need.

File: packages/react-ui-rent/src/LeadForm/LeadForm.js [hunk 8]
```diff
@@ -74,17 +75,23 @@ export default class LeadForm extends Component {
...
+      fields,
+      messagePlaceholder,
+    } = this.props
 
     return fields.map(fieldComposition => {
       const { field, ...props } = fieldComposition
       const name = props.name
       const FormField = field || FIELD_MAPPING[name] || Field
+      const isMessage = name === 'message'
+      const placeholder = isMessage ? { placeholder: messagePlaceholder } : {}
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadForm/LeadForm.js]: I'd rather put this logic in the Message component if it makes sense.

File: packages/react-ui-rent/src/LeadForm/Fields/RequiredField.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    const {
+      theme,
+      placeholder,
+      ...props
+    } = this.props
+
+    return (
+      <Field
+        className={theme.RequiredField}
+        theme={theme}
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadForm/Fields/RequiredField.js]: You don't need to pass this down.

File: packages/react-ui-rent/src/LeadForm/Fields/MoveInDate.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    name: PropTypes.string,
+    type: PropTypes.string,
+    placeholder: PropTypes.string,
+  }
+
+  static defaultProps = {
+    name: 'date',
+    type: 'date',
+    placeholder: 'Move in date',
+    theme: {},
```
Comment (tadjohnston) [inline packages/react-ui-rent/src/LeadForm/Fields/MoveInDate.js]: This prop isn't needed here.

**Comment by lfender6445** [comment]
👍

File: packages/react-ui-rent/src/LeadForm/Fields/RequiredField.js (hunk 9 above)
Comment (tcwilson) [reply packages/react-ui-rent/src/LeadForm/Fields/RequiredField.js]: I do in this case. `props` vs `this.props`. The `placeholder` is filtered out above and theme is used in `className`

**Comment by tcwilson** [comment]
@tadjohnston want to give it another shot?

File: packages/react-ui-rent/src/LeadForm/Fields/RequiredField.js (hunk 9 above)
Comment (tadjohnston) [reply packages/react-ui-rent/src/LeadForm/Fields/RequiredField.js]: That's fine you still don't need it.  Field is already themed.
